docs/seed-image-generation.md

```diff
@@ -0,0 +1,329 @@
+
+## Seed SNO Pre-Requisites
+
+The seed SNO configuration has some pre-requisites:
+
+- The CPU topology must align with the target SNO(s).
+  - Same number of cores.
+  - Same tuned performance configuration (ie. reserved CPUs).
+  - FIPS enablement (? *TODO*: confirm)
+- Deployed in the same manner as target SNO(s).
+  - Same ACM/MCE version.
+    - Exception: Hub for seed SNO must not have extra ACM addons enabled (ie. observability). The LCA orchestration confirms that no such addons are present on the seed SNO as part of its system config validation.
+- OADP operator must be deployed.
+- Container storage must be setup as shared between stateroots, such as with a separate partition.
+- Required dnsmasq configuration to support updating cluster name, domain, and IP from the seed image as part of IBU.
+
+### Shared Container Storage
+
+Image Based Upgrade (IBU) requires that container storage (`/var/lib/containers`) is setup to be shared between stateroots. This can be done by creating a separate partition for container storage when the node is installed. Please see this [blog article](https://cloud.redhat.com/blog/a-guide-to-creating-a-separate-disk-partition-at-installation-time) for more information on how this is done using a `MachineConfig`.
+
+With ZTP GitOps, the `MachineConfig` can be added as `extra-manifests` under the site-config. For example:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-containers-partitioned
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      disks:
+        - device: /dev/disk/by-id/wwn-0x62cea7f04d10350026c6f2ec315557a0
+          partitions:
+            - label: varlibcontainers
+              startMiB: 250000 # Leave room for rootfs
+              sizeMiB: 0 # Use available space
+      filesystems:
+        - device: /dev/disk/by-partlabel/varlibcontainers
+          format: xfs
+          mountOptions:
+            - defaults
+            - prjquota
+          path: /var/lib/containers
+          wipeFilesystem: true
+    systemd:
+      units:
+        - contents: |-
+            # Generated by Butane
+            [Unit]
+            Before=local-fs.target
+            Requires=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service
+            After=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service
+
+            [Mount]
+            Where=/var/lib/containers
+            What=/dev/disk/by-partlabel/varlibcontainers
+            Type=xfs
+            Options=defaults,prjquota
+
+            [Install]
+            RequiredBy=local-fs.target
+          enabled: true
+          name: var-lib-containers.mount
+```
+
+### Required dnsmasq Configuration
```

Comment:
After generating these files we have to now apply it to seed image (with or without PGT) and then finally launch the seedgen yeah?

Reply:
You need to either include this as site-policy in your gitops or manually apply the machineconfig. Otherwise, the IBU will not succeed, which is why this is now included as a validation test if you use the seed gen orchestration, and why I include this reference here. Once it is in a GA rhacm release, we can remove this and specify that rhacm release as a minimum requirement.